src/ja/server/web/requests.py

```diff
@@ -0,0 +1,111 @@
+from abc import ABC, abstractmethod
+from ja.common.message.base import Serializable
+from ja.server.database.database import ServerDatabase
+from typing import Dict
+
+
+class WebRequest(Serializable, ABC):
+    """
+    A base class for the different statistics requests.
+    """
+
+    @abstractmethod
+    def generate_report(self, database: ServerDatabase) -> None:
+        """!
+        Generate the requested statistics from the data in the database.
+        """
+
+
+class WorkMachineWorkloadRequest(WebRequest):
+    """
+    Generates the response to the machine workload request.
+    """
+
+    def __init__(self, uid: str):
+        """!
+        Initialize the request response.
+
+        @param uid The uid of the work machine this request is for.
+        """
+
+    def generate_report(self, database: ServerDatabase) -> None:
+        pass
+
+    def to_dict(self) -> Dict[str, object]:
+        pass
+
+
+class JobInformationRequest(WebRequest):
+    """
+    Generates the response to the job information request.
+    """
+
+    def __init__(self, uid: str):
+        """!
+        Initialize the request response.
+
+        @param uid The uid of the job the report is for.
```

Comment:
Can this be None to get a report for all jobs?

Reply:
Again, I don't think so.